### PR TITLE
Added input normalization parameter and fixed data snooping in hyperparameter optimization

### PR DESCRIPTION
**Pull request by eugenioLR**
This pull request adds a new parameter to the ```HundredHammers``` class that specifies the normalization strategy that will be used.

Aditionally, the issue #20 is adressed here too, since we needed to move the train/test to the beginning either way because we are supposed to normalize using only the training data set.

### CODE REVIEW

**Comment by vgarciasc** [comment]
Great work!!! I'm surprised by how cleanly the changes manage to tackle both normalization and data snooping at the same time. I like the usage of Pipeline in order to output a trained model with normalization. I changed a couple of (minor) things:

- There was an issue where `optim_hyper` was (silently) not working. Basically, it is not enough to simply pass a function as a `scoring` parameter to "GridSearchCV"; this function must be in sklearn's "scorer" format, which can be done by calling `make_scorer`. To the best of my knowledge, this happens because if GridSearchCV needs to know if the function must be minimized or maximized, and `make_scorer` looks at the function name (yes, really) to figure out this info. Since we were not calling `make_scorer`, GridSearchCV did not know how to call the function and silently did... nothing. All mean_scores were NaN and all parameter sets tied in the first place, which led to the remaining code to simply return the first `best_params` in the dataframe. I'm reporting this here with some detail because it took me a while to figure this out; indeed, I only figured out that there *was* a problem because I ran the example jupyter notebooks and was quite puzzled with the fact that the results with optim_hyper were *worse* than the results without it. Now everything's fine, I think.
- I removed the TODO: now `seed_train_test` is a HH parameter. This could be changed in the future if we want to revert to the previous framework (where training and testing was repeated with different seeds, and an average was taken);
- `n_folds` wasn't being used anymore so I took it out.

Take a look to see if I botched something up. If there's no problem, feel free to merge and close.

**Comment by eugenioLR** [comment]
I run some tests and found a small issue in the definition of the default models. Everything seems ok now.
